From dc22786980a05129c5971e68ae37b1a9f76f882d Mon Sep 17 00:00:00 2001
From: James Falcon <therealfalcon@gmail.com>
Date: Fri, 17 Sep 2021 16:25:22 -0500
Subject: [PATCH] Set Azure to apply networking config every BOOT (#1023)

In #1006, we set Azure to apply networking config every
BOOT_NEW_INSTANCE because the BOOT_LEGACY option was causing problems
applying networking the second time per boot. However,
BOOT_NEW_INSTANCE is also wrong as Azure needs to apply networking
once per boot, during init-local phase.

Signed-off-by: Henry Beberman <henry.beberman@microsoft.com>

---
 cloudinit/sources/DataSourceAzure.py                |  6 +++++-
 tests/integration_tests/modules/test_user_events.py | 10 ++++++----
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/cloudinit/sources/DataSourceAzure.py b/cloudinit/sources/DataSourceAzure.py
index 3fb564c8dd..f8641dfd2f 100755
--- a/cloudinit/sources/DataSourceAzure.py
+++ b/cloudinit/sources/DataSourceAzure.py
@@ -22,7 +22,7 @@
 from cloudinit import dmi
 from cloudinit import log as logging
 from cloudinit import net
-from cloudinit.event import EventType
+from cloudinit.event import EventScope, EventType
 from cloudinit.net import device_driver
 from cloudinit.net.dhcp import EphemeralDHCPv4
 from cloudinit import sources
@@ -339,6 +339,10 @@ def temporary_hostname(temp_hostname, cfg, hostname_command='hostname'):
 class DataSourceAzure(sources.DataSource):
 
     dsname = 'Azure'
+    default_update_events = {EventScope.NETWORK: {
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.BOOT,
+    }}
     _negotiated = False
     _metadata_imds = sources.UNSET
     _ci_pkl_version = 1
